### PR TITLE
rename Channel to Mpmc

### PR DESCRIPTION
**Pull request by JonathanStarup**
Metioned in #2710

### CODE REVIEW

File: main/test/ca/uwaterloo/flix/library/TestMpmc.flix [hunk 1]
```diff
@@ -16,65 +16,65 @@
  */
 
 @test
-def testNew01(): Channel.Mpmc & Impure = Channel.new(0)
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestMpmc.flix:19]: But I think these should belong in `Channel`...

Comment (JonathanStarup) [reply main/test/ca/uwaterloo/flix/library/TestMpmc.flix:19]: The tests? or new?

**Comment by JonathanStarup** [comment]
Channel/Mpmc should be included in prelude right? since its required for programs to compile. But this would mean that option is also required

**Comment by magnus-madsen** [comment]
> Channel/Mpmc should be included in prelude right? since its required for programs to compile. But this would mean that option is also required

Prelude? What do you mean? 

What is it done now?

**Comment by JonathanStarup** [comment]
> Prelude? What do you mean?
> 
> What is it done now?

Right now I put Mpmc in the core library. But then it doesn't compile because of its dependency to Option.

**Comment by magnus-madsen** [comment]
I don't think Channel has to be included in the smallest set. The Datalog engine is not.

**Comment by JonathanStarup** [comment]
Oh okay, what is the deciding factor on what goes in there?

**Comment by magnus-madsen** [comment]
In the minimal one, as little as possible.

**Pull request closed without merging**